### DISCUSSION TITLE
Removing right-padding if no profile is used

### DISCUSSION THREAD

**Discussion by maul0r** [mmistakes/minimal-mistakes#2449] [Q&A]
Hi, 
First, thanks for providing the theme, it looks awesome, and ou are clearly commited to it - thanks! 
I would like to know if it is possible that when posts do not display the profile (_config.yml) if the right padding for the profile area could be removed. 

If it isn't then everything is scewed to the right and it looks a bit strange. 

I have experiemented with overriding the values below in the main.scss based on your suggestions on how to change font-size. 

```
$right-sidebar-width-narrow: 200px !default;
$right-sidebar-width: 300px !default;
$right-sidebar-width-wide: 400px !default;

``` 
However, when I change those of course also pages with profile (like about) will be affected. I'd like to avoid this unintended consequence. Do you have a tip for me? My page is hosted on github pages.

**Comment by mmistakes** [comment]
Have you tried making all those values `0` ?

**Comment by maul0r** [comment]
tried two things: 
1st, ananlog to your font-size change tipp I manipulated main.scss
```
  .page {
    @include breakpoint($large) {
      float: right;
      width: calc(100% - #{0px});
      padding-right: 0px;
    }
  
    @include breakpoint($x-large) {
      width: calc(100% - #{0px});
      padding-right: 0px;
    }
}
```
Result: On Pages were profiles are wanted, like the about page, the text moves right above the footer. 

![pic](https://user-images.githubusercontent.com/30387863/77146797-2fe93600-6a8c-11ea-9013-76c3d358f56b.png)

I then imported the sacss folder and changed the variables, which results in no profiles available anywhere, presumably because there is no space for it. 

I would like, that if a profile is enabled, the appropriate padding is configured and when profile is set to false padding is removed. 

Is that possible?

**Comment by mmistakes** [comment]
Not possible by altering Sass variables alone. You'll need to hack the stylesheets to remove the negative padding and `calc` widths on both the sidebars and main content area.

They're positioned now using floats. The main content has `float: right`, that's why when there is no author sidebar there's a space on the left.

And when there is no right sidebar, there's space on the right left for it, unless you use `classes: wide` in the YAML front matter for that post/page.

The theme was designed to have sidebars, or be centered when not. If you want the main content to span the width of the screen you'll have to go custom. Search through the closed issues as there are others who have wanted to do something similar. I know I've seen customized versions of the theme that do it too... just don't remember where off-hand.
